RecoBTag/ONNXRuntime/plugins/PAIReDONNXJetTagsProducer.cc

```diff
@@ -0,0 +1,480 @@
+                                                     "pf_cand_d0_err",
+                                                     "pf_cand_dz",
+                                                     "pf_cand_dz_err",
+                                                     "pf_cand_eta_rel_1",
+                                                     "pf_cand_phi_rel_1",
+                                                     "pf_cand_eta_rel_2",
+                                                     "pf_cand_phi_rel_2"};
+  std::vector<std::string> pf_cand_vector_names_ = {"pf_cand_px", "pf_cand_py", "pf_cand_pz", "pf_cand_e"};
+  std::vector<std::string> sv_vector_names_ = {"sv_px", "sv_py", "sv_pz", "sv_e"};
+  std::vector<std::string> sv_feature_names_ = {
+      "sv_charge", "sv_chi2", "sv_d3",        "sv_d3_sig",    "sv_d0",        "sv_d0_sig",   "sv_eta",
+      "sv_mass",   "sv_ndof", "sv_ntracks",   "sv_p_angle",   "sv_phi",       "sv_pt",       "sv_x",
+      "sv_y",      "sv_z",    "sv_eta_rel_1", "sv_eta_rel_2", "sv_phi_rel_1", "sv_phi_rel_2"};
+  FloatArrays data_;  // initialize actual input array
+  // define loose cuts on jets to save space
+  float jet_pt_cut = 20;  // note: uses raw pt
+  float jet_eta_cut = 2.5;
+};
+
+PAIReDONNXJetTagsProducer::PAIReDONNXJetTagsProducer(const edm::ParameterSet& iConfig, const ONNXRuntime* cache)
+    : name_(iConfig.getParameter<std::string>("name")),
+      name_pf_(iConfig.getParameter<std::string>("name_pf")),
+      name_sv_(iConfig.getParameter<std::string>("name_sv")),
+      jet_token_(consumes<edm::View<pat::Jet>>(iConfig.getParameter<edm::InputTag>("jets"))),
+      cand_token_(consumes<reco::CandidateView>(iConfig.getParameter<edm::InputTag>("candidates"))),
+      gen_particle_token_(consumes<edm::View<reco::GenParticle>>(iConfig.getParameter<edm::InputTag>("gen_particles"))),
+      vtx_token_(consumes<VertexCollection>(iConfig.getParameter<edm::InputTag>("vertices"))),
+      sv_token_(consumes<SVCollection>(iConfig.getParameter<edm::InputTag>("secondary_vertices"))) {
+  produces<nanoaod::FlatTable>(name_);
+}
+
+void PAIReDONNXJetTagsProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("name", "PAIReDJets");
+  desc.add<std::string>("name_pf", "PAIReDPF");
+  desc.add<std::string>("name_sv", "PAIReDSV");
+  desc.add<edm::InputTag>("jets", edm::InputTag("slimmedJetsPuppi"));
+  desc.add<edm::InputTag>("candidates", edm::InputTag("packedPFCandidates"));
+  desc.add<edm::InputTag>("gen_particles", edm::InputTag("prunedGenParticles"));
+  desc.add<edm::InputTag>("vertices", edm::InputTag("offlineSlimmedPrimaryVertices"));
+  desc.add<edm::InputTag>("secondary_vertices", edm::InputTag("slimmedSecondaryVertices"));
+  desc.add<edm::FileInPath>("model_path", edm::FileInPath("RecoBTag/Combined/data/PAIReD/model3.onnx"));
+  descriptions.add("PAIReDJetTable", desc);
```

Comment:
would be nice to use `descriptions.addWithDefaultLabel(desc)` to generate the default cfi